src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+                    rospy.logerr('ObjectGoToTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during go_to roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                """ This is for tracking
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not close enough to the quad'),)
+                """
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+                roomba_velocity = math.sqrt(roomba_x_velocity**2 + roomba_y_velocity**2)
+
+                x_overshoot = roomba_x_velocity/roomba_velocity * self._overshoot
+                y_overshoot = roomba_y_velocity/roomba_velocity * self._overshoot
+
+                # p-controller
+                x_vel_target = ((self._roomba_point.point.x + x_overshoot)
+                                    * self._K_X + roomba_x_velocity)
+                y_vel_target = ((self._roomba_point.point.y + y_overshoot)
+                                    * self._K_Y + roomba_y_velocity)
+
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target =  math.copysign(self._MAX_Z_VELOCITY, z_vel_target)
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                if self._current_velocity is None:
+                    self._current_velocity = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(self._current_velocity, desired_vel)
```

Comment:
if using the `TranslateStopPlanner` you wouldn't need to limit the acceleration, it should do that for you.